Update watchOS and visionOS to officially supported (#36219)
DataDog/documentation: content/en/real_user_monitoring/application_monitoring/ios/supported_versions.md

Reviewer: ```suggestion
The RUM iOS SDK supports the following operating systems and versions of the Apple ecocosystem:
```

Reviewer 2: ```suggestion
The RUM iOS SDK supports the following platforms and versions:
```

Reviewer: Should we mention iPadOS?

Reviewer 2: ```suggestion
```

Reviewer 2: ```suggestion
The iOS SDK supports the following dependency managers:
```

---

> title: iOS, tvOS, watchOS, and visionOS Monitoring Supported Versions

Reviewer: Should we mention macOS too?

> title: iOS, tvOS, watchOS, and visionOS Monitoring Supported Versions

Reviewer 2: ```suggestion
title: Apple Platform Monitoring Supported Versions
```

> beta: true

Reviewer 2: ```suggestion
```

---
title: Apple Platform Monitoring Supported Versions

> description: "List of supported operating systems and platforms for the RUM iOS SDK."

Reviewer 2: ```suggestion
description: "Supported operating systems and platforms for the Datadog iOS SDK, including iOS, iPadOS, tvOS, watchOS, and visionOS."
```

## Supported versions

The RUM iOS SDK supports the following platforms and versions:

| Platform | Supported | Version | Notes |
|--------|-------------|---------|-------|
| iOS | {{< X >}} | 12+ | |

> | iPadOS | {{< X >}} | 12+ | |

Reviewer 2: ```suggestion
| iPadOS | {{< X >}} | 12+ | |
| tvOS | {{< X >}} | 12+ | |
| visionOS | {{< X >}} | 1.0+ | |
| watchOS | {{< X >}} | 7.0+ | |
```

Reviewer 2: sorry, made these suggestions so that they are in order of fully supported, partially supported, and not supported

| macOS (Designed for iPad) | {{< X >}} | 11+ | |
| macOS (Catalyst) | partially supported | 12+ | Catalyst is supported in build mode only, which means that macOS targets build, but functionalities for the SDK might not work for this target. |
| macOS | | 12+ | macOS is not officially supported by the Datadog SDK. Some features may not be fully functional. **Note**:  `DatadogRUM`, `DatadogSessionReplay`, and `DatadogObjc`, which heavily depend on `UIKit`, do not build on macOS. |

> | tvOS | {{< X >}} | 12+ | |

Reviewer 2: ```suggestion
```

> | visionOS | {{< X >}} | 1.0+ | |

Reviewer 2: ```suggestion
```

> | watchOS | {{< X >}} | 7.0+ | |

Reviewer 2: ```suggestion
```

| macOS (Designed for iPad) | {{< X >}} | 11+ | |
| macOS (Catalyst) | partially supported | 12+ | Catalyst is supported in build mode only, which means that macOS targets build, but functionalities for the SDK might not work for this target. |
| macOS | | 12+ | macOS is not officially supported by the Datadog SDK. Some features may not be fully functional. **Note**:  `DatadogRUM`, `DatadogSessionReplay`, and `DatadogObjc`, which heavily depend on `UIKit`, do not build on macOS. |

> | visionOS | {{< X >}} | 1.0+ | |

Reviewer 2: ```suggestion
| tvOS | {{< X >}} | 12+ | |
| visionOS | {{< X >}} | 1.0+ | |
```
